test/Microsoft.Azure.Relay.UnitTests/RunTimeTests.cs

```diff
@@ -239,12 +240,12 @@ async Task CheckBadHeaderAsync(HybridConnectionClient client, string headerName,
             }
 
             var hybridConnectionClient = GetHybridConnectionClient(EndpointTestType.Unauthenticated);
-            await CheckBadHeaderAsync(hybridConnectionClient, string.Empty, string.Empty);
-            await CheckBadHeaderAsync(hybridConnectionClient, null, null);
-            await CheckBadHeaderAsync(hybridConnectionClient, "Bad\nHeader", "Value");
-            await CheckBadHeaderAsync(hybridConnectionClient, "Bad:Header", "Value");
-            await CheckBadHeaderAsync(hybridConnectionClient, "Header", "Bad\rValue");
-            await CheckBadHeaderAsync(hybridConnectionClient, "Header", "Bad\nValue");
+            await CheckBadHeaderAsync<RelayException>(hybridConnectionClient, string.Empty, string.Empty);
+            await CheckBadHeaderAsync<ArgumentNullException>(hybridConnectionClient, null, null);
```

Comment:
>ArgumentNullException [](start = 38, length = 21)

This is thrown before our API is invoked. Not testing our API instead it tests Dictionary. Remove it and undo these other changes.

Reply:
fixed in iteration 4.

---
In reply to: [279880199](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279880199) [](ancestors = 279880199)